src/models/ProfilesModel.ts

```diff
@@ -39,47 +46,108 @@ export default class ProfilesModel extends Model {
+  public readonly extraConfigPath: string
+
+  constructor (readonly root = getMinecraftRoot()) {
     super()
-    // TODO: loadConfigSync
+    this.launchProfilePath = join(this.root, LAUNCH_PROFILE)
+    this.extraConfigPath = join(this.root, EXTRA_CONFIG)
+    try {
+      this.loadLaunchProfileJson(fs.readJsonSync(this.launchProfilePath))
+    } catch (e) {
+      this.onLoadLaunchProfileFailed(e)
+    }
+
+    try {
+      this.loadExtraConfigJson(fs.readJsonSync(this.extraConfigPath))
+    } catch (e) {
+      this.onLoadExtraConfigFailed(e)
+    }
+  }
+
+  public * setJavaPath () {
+    yield remote.dialog.showOpenDialog(remote.getCurrentWindow(), {
+      title: '选择 Java',
+      message: '请选择启动 Minecraft 的 Java 8',
+      filters: [
+        { name: '可执行文件', extensions: platform() === 'win32' ? ['exe'] : [] }
+      ]
+    }, (files) => {
+      const version = getJavaVersion(files[0])
+      if (version) {
+        this.extraJson.javaPath = files[0]
+        this.saveExtraConfigJson()
+      } else {
+        // show some ui here to let user know
+      }
+    })
   }
 
-  public * setJavaPath (path: string) {
-    // TODO: Check java exists
-    yield* this.saveConfig()
+  public * loadAllConfigs () {
+    yield fs.readJson(this.launchProfilePath).then(j => this.loadLaunchProfileJson(j),
+      e => this.onLoadLaunchProfileFailed(e))
+    yield fs.readJson(this.extraConfigPath).then(j => this.loadExtraConfigJson(j),
+      e => this.onLoadExtraConfigFailed(e))
   }
 
-  public * loadConfig () {
-    // TODO:fs-extra.readJson()
+  public * saveLaunchProfileJson () {
+    yield fs.writeJson(this.launchProfilePath, {
+      settings: this.selectedUser,
+      selectedUser: this.selectedUser,
+      profile: this.profiles,
+      authenticationDatabase: this.authenticationDatabase,
+      clientToken: this.clientToken
+    })
   }
 
-  public * saveConfig () {
-    // TODO:Async!!!!
+  public * saveExtraConfigJson () {
+    yield fs.writeJson(this.extraConfigPath, this.extraJson)
   }
 
   public * setMemory (mem: string) {
     const m = parseInt(mem, 10)
     this.extraJson.memory = Number.isNaN(m) || Object.is(m, Infinity) || m < 0 ? 0 : m
-    yield* this.saveConfig()
+    yield* this.saveExtraConfigJson()
   }
 
   public * toggleBmclAPI () {
     this.extraJson.bmclAPI = !this.extraJson.bmclAPI
-    yield* this.saveConfig()
+    yield* this.saveExtraConfigJson()
   }
 
   public * setArgs (args: string) {
     this.extraJson.javaArgs = args
-    yield* this.saveConfig()
+    yield* this.saveExtraConfigJson()
   }
 
   public * toggleSound () {
     this.settings.soundOn = !this.settings.soundOn
-    yield* this.saveConfig()
+    yield* this.saveLaunchProfileJson()
   }
 
   public * toggleShowLog () {
     this.settings.showGameLog = !this.settings.showGameLog
-    yield* this.saveConfig()
+    yield* this.saveLaunchProfileJson()
+  }
+
+  private loadLaunchProfileJson (json: any) {
+    this.selectedUser = json.selectedUser
```

Comment:
Here I directly assign the json. Though this might cause problems if the json is not valid.
We might want to use ajv or any other methods to validate the json object first.